pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -241,6 +241,29 @@ func (r *ReconcileOperConfig) Reconcile(request reconcile.Request) (reconcile.Re
 		Name:     names.APPLIED_NAMESPACE,
 	})
 
+	// Add operator.openshift.io/v1/network to relatedObjects for must-gather
+	relatedObjects = append(relatedObjects, configv1.ObjectReference{
+		Group:    "operator.openshift.io",
+		Resource: "networks",
+		Name:     "cluster",
+	})
```

Comment:
This _does_ prevent the network config from being deleted, but I think that's the wrong fix; what if someone upgrades to this version of 4.6 but then has to downgrade back to an earlier 4.6 for some reason?

In 4.5 and 4.6, we should add the hack to never delete the operator config, but we shouldn't add the operator config to `relatedObjects`. (And we should maybe add a comment warning that it's not safe to add new objects in 4.5 and 4.6.)

Reply:
> what if someone upgrades to this version of 4.6 but then has to downgrade back to an earlier 4.6 for some reason?

True as well. I will change and only keep the "never delete hack" + comment.